Project2015To2017.MigrateXXXX.Tool/CommandLogic.cs

```diff
@@ -0,0 +1,131 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using DotNet.Globbing;
+using Project2015To2017.Analysis;
+using Project2015To2017.Transforms;
+using Project2015To2017.Writing;
+using Serilog;
+
+namespace Project2015To2017.Migrate2017.Tool
+{
+	public partial class CommandLogic
+	{
+		private readonly PatternProcessor globProcessor = (converter, pattern, callback, self) =>
+		{
+			Log.Verbose("Falling back to globbing");
+			self.DoProcessableFileSearch();
+			var glob = Glob.Parse(pattern);
+			Log.Verbose("Parsed glob {Glob}", glob);
+			foreach (var (path, extension) in self.Files)
+			{
+				if (!glob.IsMatch(path)) continue;
+				var file = new FileInfo(path);
+				callback(file, extension);
+			}
+
+			return true;
+		};
+
+		private readonly MigrationFacility facility;
+
+		public CommandLogic()
+		{
+			var genericLogger = new Serilog.Extensions.Logging.SerilogLoggerProvider().CreateLogger(nameof(Serilog));
+			facility = new MigrationFacility(genericLogger, globProcessor);
+		}
+
+		public void ExecuteEvaluate(
+			IReadOnlyCollection<string> items,
+			ConversionOptions conversionOptions,
+			ITransformationSet transformationSet,
+			AnalysisOptions analysisOptions)
+		{
+			facility.ExecuteEvaluate(items, conversionOptions, transformationSet, analysisOptions);
+		}
+
+		public void ExecuteMigrate(
+			IReadOnlyCollection<string> items,
+			bool noBackup,
+			ConversionOptions conversionOptions,
+			ITransformationSet transformationSet)
+		{
+			var writeOptions = new ProjectWriteOptions { MakeBackups = !noBackup };
+			facility.ExecuteMigrate(items, transformationSet, conversionOptions, writeOptions);
+		}
+
+		public void ExecuteAnalyze(
+			IReadOnlyCollection<string> items,
+			ConversionOptions conversionOptions,
+			AnalysisOptions analysisOptions)
+		{
+			facility.ExecuteAnalyze(items, conversionOptions, analysisOptions);
+		}
+
+		public void ExecuteWizard(
```

Comment:
This signature should be reordered to match the other ones. Make `sets` the last param in stead of the first like the other ones.